### ISSUE TITLE
bedrock: support grant access to models for FoundationModel

### ISSUE THREAD

**Issue by jlosito** [aws/aws-cdk#29855]
### Describe the feature

I would like a programmatic way to grant access to Bedrock models.

### Use Case

Before using a model, a user will need to grant access to the model. A user might want to do this programmatically within CDK.

### Proposed Solution

Something along the lines of this. It could be a wrapper around a custom resource.

```
import * as bedrock from 'aws-cdk-lib/aws-bedrock';

bedrock.FoundationModel.grantModelAccess(this, 'model-access', {
    models: [bedrock.FoundationModelIdentifier.ANTHROPIC_CLAUDE_V2],
});
```

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.137.0

### Environment details (OS name and version, etc.)

macOS 14.4.1

**Comment by pahud** [comment]
Yes, we will need the L2 for FoundationModel so we can have the grant method. Making this a p1 feature request. Please help us prioritize with 👍

**Comment by pahud** [comment]
By the way, do you see any relevant SDK methods for this action? I am looking at this [doc](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Bedrock.html) but I can't find relevant methods.

**Comment by jlosito** [comment]
I personally did not. I too tried looking. From what I could come across, it looked like the model access might be a marketplace subscription under the hood.

**Comment by TheRealAmazonKendra** [comment]
Given that there are not SDK calls for this and that this is a brand new module, I don't actually believe that we should do this at this point. The service team just released their first set of resources two weeks ago and trying to build over that with custom resources is a little premature. If they then release official resources that enables this, we'll need to deprecate our custom resource and that will cause churn.

**Comment by alvazjor** [comment]
Hi @jlosito 
Bedrock modified how Foundational Models access works on their end some time ago: *Access to all Amazon Bedrock foundation models is enabled by default with the correct AWS Marketplace permissions* [[source](https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html)]
Based on this, there wont be any additional custom support from CDK side. 

Thanks
